### ISSUE TITLE
Make Sourcetrail available in apt-get for Debian and Ubuntu

### ISSUE THREAD

**Issue by egraether** [CoatiSoftware/Sourcetrail#756]

**Comment by nurupo** [comment]
This is kind of backwards. It makes more sense to ask Debian and Ubuntu package maintainers to package Sourcetrail in their distribution than asking Sourcetrail developers to become Debian/Ubuntu packagers. [Debian has a place where you can request software to be packaged](https://wiki.debian.org/RFP), and if it gets packaged in Debian it will trickle down to Ubuntu.

**Comment by Mte90** [comment]
Right now on debian cannot be compiled because sourcetrail require libboost 1.68 instead of 1.67.
So I don't think that will be interested to integrate in their repository with this need.

Sure the request can be done but if the dependence are not available they ignore the requests.
Probably is more easy to release a deb file for the moment.

**Comment by mlangkabel** [comment]
#749 may be relevant here. 
@egraether : can't we just move the minimum required boost version to 1.67?

**Comment by egraether** [comment]
Yes, I think we can. I will test it.

**Comment by egraether** [comment]
Yes, application and tests seem to work without issues with Boost 1.67.

**Comment by aweeraman** [comment]
I'm trying to build Sourcetrail on Debian and I'm getting the following:

-- lib: /usr/lib/gcc/x86_64-linux-gnu/9/../../../x86_64-linux-gnu/libgbm.so.1
-- lib: /usr/lib/gcc/x86_64-linux-gnu/9/../../../x86_64-linux-gnu/libdrm.so
-- lib: libglapi.so
CMake Warning at cmake/linux_package.cmake:28 (message):
  libglapi.so not found
Call Stack (most recent call first):
  cmake/linux_package.cmake:89 (GetAndInstallLibrary)
  cmake/linux_package.cmake:113 (AddSharedToComponent)
  CMakeLists.txt:775 (include)

I've installed the boost and Qt dependencies which it seems to find properly.

I've available on the following location on the filesystem: /usr/lib/x86_64-linux-gnu/libglapi.so

I'm no CMake expert, but let me know what you think.

If this goes well, I'm willing to get the package built and uploaded to Debian after the holidays.

**Comment by sc0w** [comment]
@aweeraman you have tested with https://github.com/CoatiSoftware/Sourcetrail/pull/784 ?

**Comment by aweeraman** [comment]
The warning for libglapi.so still persists, but resolved a linking issue at the end, so I'm able to build and run the app. thanks.

**Comment by Piraty** [comment]
Sourcetrail's cmake files targeting linux require a big rewrite before it can be packaged easily into distros. Bundling all libraries and linking statically against boost is just not going to be acceptable.

**Comment by alex-weej** [comment]
Is there any appetite from any of the maintainers to do what @Piraty says?

**Comment by emmenlau** [comment]
I can understand @Piraty 's sentiment from the distro's perspective. But that should really not hinder (and rather advocate) that you could provide your own distro packages (or snap, if you prefer to target many distro's at once).

It would be really awesome if there was a trivial way to install on Ubuntu.

**Comment by jondo** [comment]
@emmenlau How would Snap (or the less centralized Flatpack) be better than the already available AppImage? There'd still be no way to centrally fix security leaks in a used library for the whole system by just updating the library package.

**Comment by emmenlau** [comment]
@jondo I do not think that Snap, AppImage or any other form of binary distribution is better or worse than another with respect to centrally fixing security leaks. Centrally fixing security leaks (in this context) seems more a question of bundling, not generally related to the package format. Would it maybe help to untangle the two topics? I can clearly see your topic, but I'm under the impression that this should go into a separate (possibly linked) issue.

Going back to what the OP was kindly asking:

> Make Sourcetrail available in apt-get for Debian and Ubuntu

This is what I second, it would be great if Sourcetrail would be more easy to install on Debian/Ubuntu (and possibly other distro's) in binary form. For me (and possibly many other users) that may be acceptable in a highly bundled or statically linked form, albeit that may not be what all users want (or can accept).
